storage/remote/queue_manager.go

```diff
@@ -516,9 +516,10 @@ func (s *shards) sendSamples(samples model.Samples) {
 // sendSamples to the remote storage with backoff for recoverable errors.
 func (s *shards) sendSamplesWithBackoff(samples model.Samples) {
 	backoff := s.qm.cfg.MinBackoff
+	req := ToWriteRequest(samples)
+
 	for retries := s.qm.cfg.MaxRetries; retries > 0; retries-- {
 		begin := time.Now()
-		req := ToWriteRequest(samples)
```

Comment:
Did you get anywhere with pulling the snappy & proto out of the loop too?

Reply:
Couple of issues meant that it needs a little more thought, while we could push retries into the client, this would mean the client now logs the errors, which seemed sub-optimal.

And the allocs from proto and snappy did not show up in pprof, so I deferred this change to another PR to keep this PR simple enough to make tomorrows release.

Reply:
Cool, on change at a time.

Reply:
Turns out, they do show up in pprof after these changes :) I'll try to get that done before Brian wakes up tomorrow.

Reply:
I made the change but there was not enough change in allocs but added a lot of plumbing which I don't think manadates a change here.